### ISSUE TITLE
Loss function for implementation of BMAML

### ISSUE THREAD

**Issue by pandeydeep9** [cnguyen10/few_shot_meta_learning#4]
Hi. Thank you for uploading the code for all these algorithms.

For the implementation of bayesian maml (classification problem), you are using **simple cross-entropy loss.** But, bayesian maml uses the chaser loss. Specifically, for each task, I think we have to compute the chaser and leader using SVGD and then update the global parameters using the average of the difference over multiple particles (multiple models) (This might be much tricker).
 
Did I miss something in this implementation? Correct me if I am wrong or I missed something. 

thanks,
Deep

**Comment by cnguyen10** [comment]
Hi Deep,

You are right. I only implemented the simple cross-entropy loss for BMAML. I cannot recall exactly, but I think the chaser loss can be implemented for other meta-learning methods. Hence, I decided to use cross-entropy loss to fairly compare BMAML to other meta-learning algorithms.

**Comment by pandeydeep9** [comment]
Ok. Thanks for letting me know. My understanding is that the chaser loss encourages the MAML models to be diverse which leads the predicted uncertainty to be meaningful. Basically, modifying the loss function to encourage diverse MAML models is important. This might not have been fair to Bayesian MAMl.

Still, great work and thanks for sharing the code.
Deep
